From cd8e76f0b1f47271c4059cd6c1e67e23f0e247a8 Mon Sep 17 00:00:00 2001
From: Pete Batard <pete@akeo.ie>
Date: Thu, 10 May 2012 16:44:30 +0100
Subject: [PATCH 15/28] Windows: Add INSTALL_WIN.txt

libusbx.git commit 75b1cf81db74357f55cc4d0f596ef19a6a25c5cd

[stuge: Remove trailing whitespace]
---
 INSTALL_WIN.txt |   59 +++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 59 insertions(+)
 create mode 100644 INSTALL_WIN.txt

diff --git a/INSTALL_WIN.txt b/INSTALL_WIN.txt
new file mode 100644
index 0000000..745e7e8
--- /dev/null
+++ b/INSTALL_WIN.txt
@@ -0,0 +1,59 @@
+Installation Instructions for Windows
+*************************************
+
+If you are compiling for MinGW or cygwin, please refer to the INSTALL file.
+
+If you are using Microsoft Visual Studio:
+- Open the relevant solution file in /msvc:
+  libusb.dsw for MSVC6, libusb_2005.sln for Visual Studio 2005 or 2008 or
+  libusb_2010.sln for Visual Studio 2010 or later.
+- If you want to debug the library, uncomment the ENABLE_DEBUG_LOGGING define
+  in msvc\config.h
+- Select your configuration and compile the project
+
+Note that if you are using Visual Studio Express, you may have to install the
+Windows SDK to be able to compile the 64 bit version of the library.
+
+If you are using the freely available Windows DDK/WDK (Driver Development Kit)
+- If you want to debug the library, uncomment the ENABLE_DEBUG_LOGGING define
+  in msvc\config.h
+- Open one of the relevant Free Build or Checked Build prompt for your target
+  platform
+- Navigate to the msvc\ directory where the ddk_build.cmd file is located, and
+  run 'ddk_build'
+- To produce a DLL rather than a static library, use: 'ddk_build DLL'
+- To produce a static library that uses LIBCMT[d] instead of MSVCRT[d] (/MT[d]
+  vs /MD[d] in Visual Studio) use: 'ddk_build /MT'
+
+Note that using the Windows DDK, it is possible to compile both the 32 and 64
+bit versions of the library.
+
+
+Destination directories
+***********************
+
+The 32 bit binaries compiled either from Visual Studio or the DDK are placed in
+a Win32\ directory at the root of the library
+The 64 bit binaries are placed in an x64\ directory
+
+
+Troubleshooting
+***************
+
+If the compilation process complains about missing libraries, ensure that the
+default library paths for your project points to the relevant directories.
+If needed, these libraries can be obtained by installing either the latest
+Windows SDK or the DDK (Links provided at the end of this file).
+
+
+Links
+*****
+
+Additional information related to the Windows backend:
+  https://sourceforge.net/apps/mediawiki/libusbx/index.php?title=Windows
+
+Latest Windows Driver (Development) Kit (WDK):
+  http://www.microsoft.com/downloads/details.aspx?displaylang=en&FamilyID=36a2630f-5d56-43b5-b996-7633f2ec14ff
+
+Latest Microsoft Windows SDK:
+  http://www.microsoft.com/downloads/details.aspx?displaylang=en&FamilyID=c17ba869-9671-4330-a63e-1fd44e0e2505
-- 
1.7.9.5

